build.gradle.kts

```diff
@@ -71,23 +72,24 @@ tasks {
         untilBuild.set(properties("pluginUntilBuild"))
 
         // Extract the <!-- Plugin description --> section from README.md and provide for the plugin's manifest
-        pluginDescription.set(
-            file("README.md").readText().lines().run {
-                val start = "<!-- Plugin description -->"
-                val end = "<!-- Plugin description end -->"
-
-                if (!containsAll(listOf(start, end))) {
-                    throw GradleException("Plugin description section not found in README.md:\n$start ... $end")
-                }
-                subList(indexOf(start) + 1, indexOf(end))
-            }.joinToString("\n").let { markdownToHTML(it) }
+        pluginDescription.set(provider {
+                file("README.md").readText().lines().run {
+                    val start = "<!-- Plugin description -->"
+                    val end = "<!-- Plugin description end -->"
+
+                    if (!containsAll(listOf(start, end))) {
+                        throw GradleException("Plugin description section not found in README.md:\n$start ... $end")
+                    }
+                    subList(indexOf(start) + 1, indexOf(end))
+                }.joinToString("\n").let { markdownToHTML(it) }
+            }
         )
 
         // Get the latest available change notes from the changelog file
         changeNotes.set(provider {
             with(changelog) {
                 renderItem(
-                    getOrNull(properties("pluginVersion"))
+                    getOrNull(properties("pluginVersion").get())
```

Comment:
I'm not familiar with the `getOrNull` function but is it redundant with calling `.get()`?

Reply:
`getOrNull` is a method from the `ChangelogPluginExtension` and the it will return if it doesn't find a corresponding changelog for a given version. It does not accept a provider. However, I improved it a bit.